handler/src/main/java/io/netty/handler/ssl/OpenSslKeyMaterialManager.java

```diff
@@ -79,101 +74,61 @@ void setKeyMaterial(ReferenceCountedOpenSslEngine engine) throws SSLException {
             if (type != null) {
                 String alias = chooseServerAlias(engine, type);
                 if (alias != null && aliases.add(alias)) {
-                    setKeyMaterial(ssl, alias, engine.alloc);
+                    OpenSslKeyMaterial keyMaterial = null;
+                    try {
+                        keyMaterial = provider.chooseKeyMaterial(engine.alloc, alias);
+                        SSL.setKeyMaterialServerSide(ssl, keyMaterial.chain(), keyMaterial.privateKey());
+                    } catch (SSLException e) {
+                        throw e;
+                    } catch (Exception e) {
+                        throw new SSLException(e);
+                    } finally {
+                        if (keyMaterial != null) {
+                            keyMaterial.release();
+                        }
+                    }
                 }
             }
         }
     }
 
-    CertificateRequestedCallback.KeyMaterial keyMaterial(ReferenceCountedOpenSslEngine engine, String[] keyTypes,
+    void setkeyMaterial(ReferenceCountedOpenSslEngine engine, long certOut, long keyOut, String[] keyTypes,
```

Comment:
`setkeyMaterial` -> `setKeyMaterial`? Also since this method is specific to client side should this be `setClientKeyMaterial`, and the other method be `setServerKeyMaterial`?

Reply:
Used the same naming as I used in tcnative... Good idea.